mifosng-android/src/main/java/com/mifos/mifosxdroid/online/clientidentifiers/ClientIdentifiersFragment.java

```diff
@@ -130,16 +132,34 @@ public void showUserInterface() {
     public void showClientIdentifiers(List<Identifier> identifiers) {
         this.identifiers = identifiers;
         identifierListAdapter.setIdentifiers(identifiers);
+        identifierListAdapter.notifyDataSetChanged();
+
+        if (identifiers.isEmpty()) {
+            showEmptyClientIdentifier();
+        } else {
+            if (ll_error.getVisibility() == View.VISIBLE) {
+                ll_error.setVisibility(View.GONE);
+            }
+        }
     }
 
-    @Override
-    public void showEmptyClientIdentifier() {
+    private void showEmptyClientIdentifier() {
         ll_error.setVisibility(View.VISIBLE);
         mNoIdentifierText.setText(getResources().getString(R.string.no_identifier_to_show));
         mNoIdentifierIcon.setImageResource(R.drawable.ic_assignment_turned_in_black_24dp);
     }
 
     @Override
+    public void onClientIdentifierCreationSuccess() {
+        loadIdentifiers();
```

Comment:
Why do you want to retrieve all the identifiers again, when you create a new Identifier ? You have most of the data already that you ```POST```, you'll get some stuff from the server in the response like ID or whatever that the server is responsible for generating. Why don't you just push that stuff into your list ?